packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -122,6 +123,21 @@ contract L2Genesis is Deployer {
         runWithOptions(OutputMode.ALL, LATEST_FORK, artifactDependencies());
     }
 
+    /// @notice This is used by new experimental interop deploy tooling.
+    function runWithEnv() public {
+        //  The setUp() is skipped (since we insert a custom DeployConfig, and do not use Artifacts)
```

Comment:
How is `setUp` skipped? When invoked via forge `setUp` will always be run first

Reply:
In the Go scripting we can choose to call `setUp` or not, since it's just one yet another contract function to Go.

In this case the legacy `DeployConfig` requires unrelated (non-L2) fields to be set, and the legacy `Artifacts` isn't possible to swap (due to it being a library and not a standalone, and the library directly using file IO).

So we leave these things uninitialized, and avoid them during execution (swap the deploy config with a precompile that can present the partial config that is actually used, and load the 3 L1 dependencies from the environment, rather than using the legacy Artifacts code).

It's not ideal, but it works, and avoids a large refactor of `L2Genesis.s.sol` configuration at this time. I'd like to refactor it to be a clean script, but that might break the old pre-OPSM pre-op-deployer genesis code-path, which we don't have a stable replacement for yet.

Reply:
I see, thanks